clippy_lints/src/loops.rs

```diff
@@ -1017,6 +1052,238 @@ fn detect_manual_memcpy<'tcx>(
     }
 }
 
+// Delegate that traverses expression and detects mutable variables being used
+struct UsesMutableDelegate {
```

Comment:
Can this stuff be in `utils`?

Reply:
I used [`mutated_variables`](https://github.com/rust-lang/rust-clippy/blob/5a20489c5ca0951827cbb1b4d72ddfdfd393713a/clippy_lints/src/utils/usage.rs#L14) in utils. How about this?